src/notation/internal/notationinteraction.cpp

```diff
@@ -1772,6 +1773,48 @@ bool NotationInteraction::applyPaletteElement(mu::engraving::EngravingItem* elem
                 e = toChord(e)->notes().front();
             }
             applyDropPaletteElement(score, e, element, modifiers);
+        } else if (element->type() == ElementType::MARKER
+                   || element->type() == ElementType::JUMP
+                   || element->type() == ElementType::SPACER
+                   || element->type() == ElementType::VBOX
+                   || element->type() == ElementType::HBOX
+                   || element->type() == ElementType::TBOX
+                   || element->type() == ElementType::MEASURE
+                   || element->type() == ElementType::BRACKET
+                   || (element->type() == ElementType::ACTION_ICON
+                       && (toActionIcon(element)->actionType() == mu::engraving::ActionIconType::VFRAME
+                           || toActionIcon(element)->actionType() == mu::engraving::ActionIconType::HFRAME
+                           || toActionIcon(element)->actionType() == mu::engraving::ActionIconType::TFRAME
+                           || toActionIcon(element)->actionType() == mu::engraving::ActionIconType::STAFF_TYPE_CHANGE
+                           || toActionIcon(element)->actionType() == mu::engraving::ActionIconType::MEASURE
+                           || toActionIcon(element)->actionType() == mu::engraving::ActionIconType::BRACKETS))) {
+            // we add the following measure-based items to each measure containing selected items
+            std::vector<Measure*> measuresWithSelectedContent;
+            for (EngravingItem* e : sel.elements()) {
+                Measure* m = e->findMeasure();
+                if (!m) {
+                    continue;
+                }
+                if (element->type() == ElementType::MARKER && e->isBarLine()
+                    && toBarLine(e)->segment()->segmentType() != SegmentType::BeginBarLine
+                    && toBarLine(e)->segment()->segmentType() != SegmentType::StartRepeatBarLine) {
+                    // exception: markers are anchored to the start of a measure,
+                    // so when the user selects an end barline we take the next measure
+                    m = m->nextMeasureMM() ? m->nextMeasureMM() : m;
+                }
+                if (muse::contains(measuresWithSelectedContent, m)) {
+                    continue;
+                }
+                measuresWithSelectedContent.push_back(m);
+
+                RectF r = m->staffPageBoundingRect(e->staff()->idx());
+                PointF pt(r.x() + r.width() * .5, r.y() + r.height() * .5);
+                pt += m->system()->page()->pos();
```

Comment:
The reason I left this in here is because it's present in the code for adding these elements to a range selection. I feel like it should work without, should I try removing it in both places?